### ISSUE TITLE
select all not working in select2 after setting template

### ISSUE THREAD

**Issue by araviaravi** [kartik-v/yii2-widget-select2#248]
Browser: All

Code: 
```


<?php
$format = <<< SCRIPT
function format(state) {
  if (!state.id){ return state.text;} // optgroup
  if(state.text == 'Active'){return '<span class="label label-info label-rouded m-r-5 req-status">&nbsp;</span>' + state.text;}
  if(state.text == 'Closed'){return '<span class="label label-success label-rouded m-r-5 req-status">&nbsp;</span>' + state.text;}
  if(state.text == 'Hold'){return '<span class="label label-warning label-rouded m-r-5 req-status">&nbsp;</span>' + state.text;}
}
SCRIPT;
        ?>

        <?php
          $escape = new JsExpression("function(m) { return m; }");
          $this->registerJs($format, View::POS_HEAD);
        ?>

        <?= $form->field($model,'status')->widget(Select2::classname(),[
          'data'=>[Requirement::STATUS_ACTIVE =>'Active',Requirement::STATUS_CLOSED =>'Closed', Requirement::STATUS_HOLD =>'Hold',],
          'options'=>['placeholder' => 'All Status'],
          'pluginOptions'=>[
            'templateResult' => new JsExpression('format'),
            'templateSelection' => new JsExpression('format'),
            'escapeMarkup' => $escape,
            'multiple' => true,
            'allowClear' => true,
          ],
          ])->label(false)  ?>
```

**Comment by araviaravi** [comment]
![ezgif com-crop](https://user-images.githubusercontent.com/20757822/35899609-5b15f592-0bf3-11e8-91de-5c21828a8382.gif)

**Comment by kartik-v** [comment]
Duplicate to #246 and will be resolved via #248.
